odoo/models.py

```diff
@@ -3426,76 +3427,78 @@ def unlink(self):
             if func._ondelete or not self._context.get(MODULE_UNINSTALL_FLAG):
                 func(self)
 
-        # mark fields that depend on 'self' to recompute them after 'self' has
-        # been deleted (like updating a sum of lines after deleting one line)
-        self.flush()
-        self.modified(self._fields, before=True)
-
-        with self.env.norecompute():
-            self.check_access_rule('unlink')
-
-            cr = self._cr
-            Data = self.env['ir.model.data'].sudo().with_context({})
-            Defaults = self.env['ir.default'].sudo()
-            Property = self.env['ir.property'].sudo()
-            Attachment = self.env['ir.attachment'].sudo()
-            ir_model_data_unlink = Data
-            ir_attachment_unlink = Attachment
-
-            # TOFIX: this avoids an infinite loop when trying to recompute a
-            # field, which triggers the recomputation of another field using the
-            # same compute function, which then triggers again the computation
-            # of those two fields
-            for field in self._fields.values():
+        # TOFIX: this avoids an infinite loop when trying to recompute a
+        # field, which triggers the recomputation of another field using the
+        # same compute function, which then triggers again the computation
+        # of those two fields
+        for field in self._fields.values():
+            if field.compute and field.store:
                 self.env.remove_to_compute(field, self)
 
-            for sub_ids in cr.split_for_in_conditions(self.ids):
-                # Check if the records are used as default properties.
-                refs = ['%s,%s' % (self._name, i) for i in sub_ids]
-                if Property.search([('res_id', '=', False), ('value_reference', 'in', refs)], limit=1):
-                    raise UserError(_('Unable to delete this document because it is used as a default property'))
-
-                # Delete the records' properties.
-                Property.search([('res_id', 'in', refs)]).unlink()
-
-                query = "DELETE FROM %s WHERE id IN %%s" % self._table
-                cr.execute(query, (sub_ids,))
+        self.flush()
```

Comment:
might be good to add a comment detailing what you explained in the commit message, as the commit message might be lost if this piece of code is ever moved/modified

Reply:
OTOH going through the blame history should always show the original commit message, while comments can move around (confusingly ending up linked to unrelated pieces of code), or not be updated / removed when the code is updated e.g. https://github.com/odoo/odoo/pull/68111#discussion_r600488747 where contributor got worried about a comment which turns out probably hasn't applied for a year (and that's a case where the comment was probably a good idea too).